### ISSUE TITLE
Regression: cannot order by boolean in 2.3

### ISSUE THREAD

**Issue by jonathancrosmer** [neo4j/neo4j#5745]
We had a Cypher 2.1 query that included a clause like `ORDER BY n.bool DESC` where `n.bool` is a Boolean property.  After upgrading to 2.3, the query throws an exception.  I didn't test 2.2.

```
Caused by: org.neo4j.cypher.IncomparableValuesException: Don't know how to compare that. Left: false (Boolean); Right: false (Boolean)
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.incomparableValuesException(CompatibilityFor2_3.scala:67) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.incomparableValuesException(CompatibilityFor2_3.scala:58) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.frontend.v2_3.IncomparableValuesException.mapToPublic(CypherException.scala:115) ~[org.neo4j.neo4j-cypher-frontend-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.runSafely(CompatibilityFor2_3.scala:119) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper.run(CompatibilityFor2_3.scala:191) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.PreparedPlanExecution.execute(PreparedPlanExecution.scala:27) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:118) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.ExecutionEngine.execute(ExecutionEngine.scala:104) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.javacompat.ExecutionEngine.execute(ExecutionEngine.java:87) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.springframework.data.neo4j.support.query.CypherQueryEngineImpl.parseAndExecuteQuery(CypherQueryEngineImpl.java:71) ~[org.springframework.data.spring-data-neo4j-3.4.0.RELEASE.jar:na]
    ... 41 common frames omitted
Caused by: org.neo4j.cypher.internal.frontend.v2_3.IncomparableValuesException: null
    at org.neo4j.cypher.internal.compiler.v2_3.Comparer$class.compare(Comparer.scala:38) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.TopPipe.compare(TopPipe.scala:35) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.TopPipe$LessThanComparator.compare(TopPipe.scala:50) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.TopPipe$LessThanComparator.compare(TopPipe.scala:44) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at java.util.TimSort.countRunAndMakeAscending(TimSort.java:324) ~[na:1.7.0_79]
    at java.util.TimSort.sort(TimSort.java:189) ~[na:1.7.0_79]
    at java.util.TimSort.sort(TimSort.java:173) ~[na:1.7.0_79]
    at java.util.Arrays.sort(Arrays.java:659) ~[na:1.7.0_79]
    at scala.collection.SeqLike$class.sorted(SeqLike.scala:648) ~[org.scala-lang.scala-library-2.11.7.jar:na]
    at scala.collection.mutable.ArrayOps$ofRef.sorted(ArrayOps.scala:186) ~[org.scala-lang.scala-library-2.11.7.jar:na]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.TopPipe.internalCreateResults(TopPipe.scala:96) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:125) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:122) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:122) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.pipes.PipeWithSource.createResults(Pipe.scala:122) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.createResults(DefaultExecutionResultBuilderFactory.scala:93) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.DefaultExecutionResultBuilderFactory$ExecutionWorkflowBuilder.build(DefaultExecutionResultBuilderFactory.scala:63) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:230) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anonfun$getExecutionPlanFunction$1.apply(ExecutionPlanBuilder.scala:214) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compiler.v2_3.executionplan.ExecutionPlanBuilder$$anon$3.run(ExecutionPlanBuilder.scala:172) ~[org.neo4j.neo4j-cypher-compiler-2.3-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.CompatibilityFor2_3$ExecutionPlanWrapper$$anonfun$run$1.apply(CompatibilityFor2_3.scala:192) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    at org.neo4j.cypher.internal.compatibility.exceptionHandlerFor2_3$.runSafely(CompatibilityFor2_3.scala:114) ~[org.neo4j.neo4j-cypher-2.3.0.jar:2.3.0]
    ... 47 common frames omitted
```

Workaround:

`ORDER BY (CASE my.bool WHEN TRUE THEN 0 ELSE 1 END)`

**Comment by Mats-SX** [comment]
Hello @jonathancrosmer and thanks for reporting this issue. We've found the bug and it's going to be fixed in 2.3.1.

**Comment by jonathancrosmer** [comment]
Awesome, thanks
